doc/changelog.rst

```diff
@@ -1,25 +1,35 @@
 Changelog
 =========
 
-Changes in Version 4.14.0 (XXXX/XX/XX)
+Changes in Version 4.14.0 (2025/08/06)
 --------------------------------------
+
 PyMongo 4.14 brings a number of changes including:
 
-- Added preliminary support for Python 3.14 and 3.14 with free-threading.  We do not yet support the following with Python 3.14:
-    - Subinterpreters (``concurrent.interpreters``)
-    - Free-threading with Encryption
-    - mod_wsgi
+- Dropped support for MongoDB 4.0.
```

Comment:
Could you also add a `..warning` for this like we did for MongoDB 3.6 in pymongo 4.11?